R-package/configure.win

```diff
@@ -6,7 +6,9 @@
 ###########################
 
 R_EXE="${R_HOME}/bin${R_ARCH_BIN}/R"
-CC=`"${R_EXE}" CMD config CC`
```

Comment:
Why do we keep `CC` for Unix (considering #4506 as well)?
https://github.com/microsoft/LightGBM/blob/b43b6c71d6b01e738b71aaad44aaa759f026ee7c/R-package/configure.ac#L23

Reply:
I wanted to keep this PR's scope limited to Windows.

A C compiler is only used on the unix-alike side (`configure.ac`) for the OpenMP test, but #4507 proposes changing that to a C++ compiler.

Reply:
Based on this comment thread, I just removed the changes for OpenMP checks from #4506, so now there won't be overlap between this PR, #4506, and #4507.